packages/sdk/src/services/invitation-service.ts

```diff
@@ -0,0 +1,106 @@
+import { Connection, Invitation, InvitationApi } from '@meeco/vault-api-sdk';
+import { EncryptionKey } from '../models/encryption-key';
+import Service, { IDEK, IKEK, IKeystoreToken, IVaultToken } from './service';
+
+export class InvitationService extends Service<InvitationApi> {
+  public getAPI(vaultToken: string): InvitationApi {
+    return this.vaultAPIFactory(vaultToken).InvitationApi;
+  }
+
+  public async create(
+    credentials: IVaultToken & IKeystoreToken & IDEK & IKEK,
+    name: string
+  ): Promise<Invitation> {
+    const {
+      vault_access_token,
+      keystore_access_token,
+      key_encryption_key,
+      data_encryption_key,
+    } = credentials;
+
+    const keyPair = await this.createAndStoreKeyPair(keystore_access_token, key_encryption_key);
+
+    this.logger.log('Encrypting recipient name');
+    const encryptedName: string = await this.encryptName(name, data_encryption_key);
+
+    this.logger.log('Sending invitation request');
+    return await this.getAPI(vault_access_token)
+      .invitationsPost({
+        public_key: {
+          keypair_external_id: keyPair.keystoreStoredKeyPair.id,
+          public_key: keyPair.keyPair.publicKey,
+        },
+        invitation: {
+          encrypted_recipient_name: encryptedName,
+        },
+      })
+      .then(result => result.invitation);
+  }
+
+  public async accept(
+    credentials: IVaultToken & IKeystoreToken & IKEK & IDEK,
+    name: string,
+    invitationToken: string
+  ): Promise<Connection> {
+    const {
+      keystore_access_token,
+      vault_access_token,
+      key_encryption_key,
+      data_encryption_key,
+    } = credentials;
+    const keyPair = await this.createAndStoreKeyPair(keystore_access_token, key_encryption_key);
+
+    this.logger.log('Encrypting connection name');
+    const encryptedName: string = await this.encryptName(name, data_encryption_key);
+
+    this.logger.log('Accepting invitation');
+    return await this.vaultAPIFactory(vault_access_token)
```

Comment:
My preference is to not `return await` unless you actually need to (i.e. if it's in a try-catch block so you can handle the error). You might as well just return the promise.

Reply:
My preference too! Good spot!